justfile

```diff
@@ -0,0 +1,148 @@
+# Run `just` to see all available commands
+
+# Default recipe - show help when running just without arguments
+default:
+    @just --list
+
+# Variables
+benchmark_source := "benchmarks/"
+benchmark_backends := "inductor,eager"
+benchmark_opts := ""
+
+# Ensure virtual environment is available and set up
+# Note: This recipe doesn't actually activate the venv for subsequent commands
+# because each line in a just recipe runs in a separate shell.
+# Each command must explicitly use the venv paths.
+_ensure-venv:
+    #!/usr/bin/env bash
+    if [[ ! -f "venv/bin/activate" ]]; then
+        echo "🔧 Setting up virtual environment..."
+        ./setup_dev_env.sh
+    else
+        echo "Virtual environment found!"
+    fi
+
+# Run all tests with coverage
+test-all: _ensure-venv
+    ./venv/bin/pytest -v --device all --dtype all --cov=kornia tests/
+
+# Run CPU-only tests
+test-cpu: _ensure-venv
+    ./venv/bin/pytest -v --device cpu --dtype all --cov=kornia tests/
+
+test-cpu-f32: _ensure-venv
+    ./venv/bin/pytest -v --device cpu --dtype float32 --cov=kornia tests/
+
+# Run CUDA tests
+test-cuda: _ensure-venv
+    ./venv/bin/pytest -v --device cuda --dtype all --cov=kornia tests/
+
+test-cuda-f32: _ensure-venv
+    ./venv/bin/pytest -v --device cuda --dtype float32 --cov=kornia tests/
+
+# Run MPS tests (Apple Silicon)
+test-mps: _ensure-venv
+    ./venv/bin/pytest -v --device mps --dtype float32 -k "not (grad or exception or jit or dynamo)" tests/
+
+# Run tests for a specific module (usage: just test-module tests/test_color.py)
+test-module module: _ensure-venv
+    ./venv/bin/pytest -v --device all --dtype all tests/{{module}}
+
+# Run JIT compilation tests
+test-jit: _ensure-venv
```

Comment:
as a followup task we should double which of those a still used. We want by the end of the year to make a major release with some considerable cleanup to simplify the lib maintaince